code/workspaces/auth-service/src/dataaccess/userRolesRepository.js

```diff
@@ -31,7 +31,12 @@ async function getUser(userId) {
 
 async function getUsers() {
   const allRoles = await UserRoles().find().exec();
-  return allRoles.map(roles => convertToUser(roles));
+  const users = allRoles
+    .filter(roles => roles.userName) // only take users with known user names
```

Comment:
I assume at this point the placeholder string of `'Unknown user name'` hasn't been injected?

Reply:
Yes, that happens in the next line,
`.map(roles => convertToUser(roles))
`